### PR TITLE
Feature/combine-vm-module/additions

### PR DESCRIPTION
**Pull request by jk-ans**
Merging changes back into combine-vm-module branch.

Added:
 - Accelerated Network
 - Key Vault Extention 
 - Win OS Diag 
 - AAD Login Ext
 - Option for hotpaching

Please select the option that is relevant.

  - [ ] Bug fix (non-breaking change which fixes an issue)
  - [x] New feature (non-breaking change which adds functionality)
  - [x] Breaking change (fix or feature that would cause existing functionality to not work as expected)
  - [x] This change requires a documentation update

# Checklist:

  - [x] My code follows the style guidelines of this project
  - [x] I have performed a self-review of my code
  - [x] I have commented my code, particularly in hard-to-understand areas
  - [x] I have made corresponding changes to the documentation
  - [x] My changes generate no new warnings e.g. code analysis tooling or general use of the changed code

### CODE REVIEW

File: azure/virtual-machine/diag-settings/win-diag-settings.json [hunk 1]
```diff
@@ -0,0 +1,273 @@
+{
```
Comment (djackson-ans) [inline azure/virtual-machine/diag-settings/win-diag-settings.json:1]: Do we need to do this in JSON? Could we do this in HCL and then convert it to JSON?

File: azure/virtual-machine/extensions.tf [hunk 2]
```diff
@@ -82,3 +82,82 @@ resource "azurerm_dev_test_global_vm_shutdown_schedule" "main" {
...
+  type_handler_version       = "1.0"
+  auto_upgrade_minor_version = true
+  automatic_upgrade_enabled  = false
+}
+
+#########################
+# This requires Python 2
+#########################
+
+# resource "azurerm_virtual_machine_extension" "lin-diag" {
```
Comment (djackson-ans) [inline azure/virtual-machine/extensions.tf:141]: can this commented code be removed?

File: azure/virtual-machine/examples/basic/main.tf [hunk 3]
```diff
@@ -57,7 +59,21 @@ module "vm" {
   tags                = local.tags
 
   computer_name = "vm"
-  password      = var.password
+  password      = "ASfdskjh3&^"
```
Comment (djackson-ans) [inline azure/virtual-machine/examples/basic/main.tf:62]: Could we replace this with a random generator for the example?

File: azure/virtual-machine/examples/basic/main.tf [hunk 4]
```diff
@@ -9,18 +9,20 @@ provider "azurerm" {
   }
 }
 
-variable "password" {
-  description = "VM Password."
-  type        = string
-  sensitive   = true
-}
+# variable "password" {
```
Comment (djackson-ans) [inline azure/virtual-machine/examples/basic/main.tf:12]: Can we remove this comment?

File: azure/virtual-machine/examples/basic/main.tf (hunk 3 above)
Comment (jk-ans) [reply azure/virtual-machine/examples/basic/main.tf:62]: replaced with var.password

File: azure/virtual-machine/extensions.tf (hunk 2 above)
Comment (jk-ans) [reply azure/virtual-machine/extensions.tf:141]: left as optional

File: azure/virtual-machine/variables.tf [hunk 5]
```diff
@@ -105,6 +105,12 @@ variable "enable_ip_forwarding" {
   default     = false
 }
 
+variable "enable_accelerated_networking" {
```
[the remark this answers is not in the document]
Comment (jk-ans) [reply azure/virtual-machine/variables.tf:108]: done

**Comment by mbulfin-ans** [comment]
Looks fine to me

**Review by mbulfin-ans — APPROVED** [review]
All look good

**Review by tb-ans — APPROVED** [review]

**Pull request merged**